File: RepoTransBench/target_projects/Python/C++/thomasa88_ParametricText/public_tests/test_public_parser.cpp
#include <gtest/gtest.h>
#include "paramparser.h"

TEST(PublicParser, ParamSpecNameValue) {
    ParamSpec spec("public_name", std::string("17"));
    EXPECT_EQ(spec.var, "public_name");
    ASSERT_TRUE(spec.member.has_value());
    EXPECT_EQ(spec.member.value(), "17");
}

TEST(PublicParser, ParamSpecStrAndEq) {
    ParamSpec spec1("ab", std::string("9"));
    ParamSpec spec2("ab", std::string("9"));
    ParamSpec spec3("ab", std::string("8"));
    std::string expected = "(ab, 9)";
    // .to_string() begins with expected
    EXPECT_EQ(spec1.to_string().substr(0, expected.size()), expected);
    EXPECT_TRUE(spec1 == spec2);
    EXPECT_FALSE(spec1 == spec3);
}

File: RepoTransBench/target_projects/Python/Rust/thomasa88_ParametricText/public_tests/test_public_parser.rs
use parametric_text::paramparser::{ParamSpec};

#[test]
fn test_ParamSpec_name_value() {
    let spec = ParamSpec::new("public_name", Some("17"), None, None);
    assert_eq!(spec.var, "public_name");
    assert_eq!(spec.member.as_deref(), Some("17"));
}

#[test]
fn test_ParamSpec_str_and_eq() {
    let spec1 = ParamSpec::new("ab", Some("9"), None, None);
    let spec2 = ParamSpec::new("ab", Some("9"), None, None);
    let spec3 = ParamSpec::new("ab", Some("8"), None, None);
    assert_eq!(spec1.to_string(), "(ab, 9)");
    assert_eq!(spec1, spec2);
    assert_ne!(spec1, spec3);
}